src/hydra/dataProvider.js

```diff
@@ -170,7 +170,9 @@ export default (
     }
 
     return convertReactAdminDataToHydraData(resource, data).then(data =>
-      JSON.stringify(data),
+      undefined === resource.encodeData
+        ? JSON.stringify(data)
+        : resource.encodeData(data),
```

Comment:
could you add a test? I'd do the opposite btw:

```
resource.encodeData ? resource.encodeData(data) : JSON.stringify(data)
```

Reply:
I agree for the code, I'd just restored the original one.

For the test, I try to find the right solution first